core/src/main/resources/db/changelog/changesets/2023_05_23_refactor_tdr_tables.yaml

```diff
@@ -0,0 +1,19 @@
+databaseChangeLog:
+  - changeSet:
+      id: "refactor_tdr_tables"
+      author: mbemis
+      changes:
+        - renameColumn:
+            oldColumnName: dataset_id
+            newColumnName: tdr_dataset_id
+            tableName: dataset
+        - addColumn:
+            tableName: data_repo_job
+            columns:
+              - column: { name: dataset_id, type: uuid, constraints: { nullable: false } }
+        - addForeignKeyConstraint:
+            baseTableName: data_repo_job
```

Comment:
is there a reason why you have this constraint declared in a separate block from the one above? 
adding `foreignKeyName: fk_dataset_job_dataset_id, references: dataset(id)` inside the "constraints" block above should accomplish the same thing and keep the schema easier to read